modules/tensor_mechanics/include/userobjects/CrackFrontPointsProvider.h

```diff
@@ -0,0 +1,58 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef CRACKFRONTPOINTSPROVIDER_H
+#define CRACKFRONTPOINTSPROVIDER_H
+
+#include "GeneralUserObject.h"
+
+class CrackFrontPointsProvider;
+
+template <>
+InputParameters validParams<CrackFrontPointsProvider>();
+
+/**
+ * Base class for crack front points provider
+ */
+class CrackFrontPointsProvider : public GeneralUserObject
+{
+public:
+  CrackFrontPointsProvider(const InputParameters & parameters);
+
+  virtual void initialSetup() override{};
+  virtual void initialize() override{};
+  virtual void execute() override{};
+  virtual void finalize() override{};
+
+  /**
+   *
+   */
+  virtual void calculateCrackFrontPoints(){};
+
+  /**
+   *
+   */
+  virtual const std::vector<Point> & getCrackFrontPoints() const { return _crack_front_points; };
```

Comment:
I don't really see the need to have both getCrackFrontPoints and calculateCrackFrontPoints. I'd keep getCrackFrontPoints and get rid of the other one.